src/inference/src/dev/core_impl.cpp

```diff
@@ -338,20 +338,16 @@ ov::SoPtr<InferenceEngine::IExecutableNetworkInternal> ov::CoreImpl::compile_mod
         res = load_model_from_cache(cacheContent, plugin, parsed._config, {}, loadedFromCache);
         if (!loadedFromCache) {
             res = compile_model_impl(model, plugin, parsed._config, {}, cacheContent, forceDisableCache);
-        } else {
-            // Temporary workaround until all plugins support caching of original model inputs
-            InferenceEngine::SetExeNetworkInfo(res._ptr, model, is_new_api());
```

Comment:
is it possible that in case of old Inference Engine API we will loose some information after removing this line?

Reply:
This code works only with new representation, as showed tests it works.